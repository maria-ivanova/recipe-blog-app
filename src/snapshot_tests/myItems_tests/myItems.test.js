import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import Routing from '../helpers/router.js';
import MyItems from '../../components/myItems.js';

import { library } from '@fortawesome/fontawesome-svg-core';
import { faUser, faHeart } from '@fortawesome/free-solid-svg-icons';
library.add(faUser, faHeart);

describe('My items page Component', () => {
    it('should render layout missing items', () => {
        const tree = renderer.create(
            <Routing value={{ displayName: "test", email: "test@test.com", uid: "1" }}>
                <MyItems />
            </Routing>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render layout with items', () => {
        const wrapper = mount(shallow(
            <Routing value={{ displayName: "test", email: "test@test.com", uid: "1" }}>
                <MyItems />
            </Routing>
        ).get(0))

        let component = wrapper.find('MyItems');
        component.setState({
            data: [
                {
                    category: "test category",
                    createdDate: 1597128798230,
                    creatorId: "1",
                    creatorName: "user1",
                    id: "id1",
                    imageUrl: "test url",
                    ingredients: "test ingredients",
                    likes: 1,
                    likesArr: ["id1"],
                    recipeDescription: "test description",
                    title: "test title",
                    totalTime: "30 мин.",
                    yields: "1",
                }
            ]
        });
        component = component.update();

        expect(component.find('.items_list .item').length).toBe(1);
    })

    it('test delete item', () => {
        const wrapper = mount(shallow(
            <Routing value={{ displayName: "test", email: "test@test.com", uid: "1" }}>
                <MyItems />
            </Routing>
        ).get(0))

        let component = wrapper.find('MyItems');
        component.setState({
            data: [
                {
                    category: "test category",
                    createdDate: 1597128798230,
                    creatorId: "1",
                    creatorName: "user1",
                    id: "id1",
                    imageUrl: "test url",
                    ingredients: "test ingredients",
                    likes: 1,
                    likesArr: ["id1"],
                    recipeDescription: "test description",
                    title: "test title",
                    totalTime: "30 мин.",
                    yields: "1",
                }
            ]
        });

        component.update().find('.items_list .btn_del').simulate('click');
        expect(component.find('.items_list .item').length).toBe(0);
      })

    
})
