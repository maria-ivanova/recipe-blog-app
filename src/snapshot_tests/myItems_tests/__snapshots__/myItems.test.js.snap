// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My items page Component should render layout missing items 1`] = `
Array [
  <div
    className="notifications 
                             null 
                             "
  >
    <span>
      
    </span>
  </div>,
  <section
    className="sec"
  >
    <div
      className="banner_item"
    >
      <h2
        className="title"
      >
        Моите рецепти
      </h2>
    </div>
  </section>,
  <section
    className="sec content_sec"
  >
    <div
      className="container"
    >
      <div
        className="items_list my_list"
      >
        <div
          className="inner_text"
        >
          <h3>
            Все още нямате създадени рецепти!
          </h3>
          <a
            className="btn btn"
            href="/create"
            onClick={[Function]}
          >
            Създай рецепта
          </a>
        </div>
      </div>
    </div>
  </section>,
]
`;
